### ISSUE TITLE
BDR: If 'max_replication_slots' is too low for the size of the cluster, the user is not notified

### ISSUE THREAD

**Issue by kforte90** [2ndQuadrant/bdr#65]
If the user sets 'max_replication_slots' to a lower value than the number of nodes being added to a cluster, there is no notification that the cluster will be ineffective.  In a negative, test, one sees that new nodes starting at 'max_replication_slots +1' will not be part of the BDR group.
It looks as if when a node is removed, the table 'pg_replication_slots' remains static in size, this may be by design.
The user should set the GUC variable 'max_replication_slots' to a high value for future cluster growth as there doesn't seem to be a trade-off in terms of system resource tradeoffs.

**Comment by ringerc** [comment]
If replication slots aren't dropped on node remove, that's a bug, and one
I'm surprised hasn't come up before.

I'll check the testcases.

**Comment by ringerc** [comment]
Yep, we don't drop the slot. Additional regression test added, then I'll
fix the bug.​

**Comment by ringerc** [comment]
The slot drop is fixed in #73, Andres implemented it while I was still trying to get the tests for my implementation working.

**Comment by ringerc** [comment]
To report an error we'd have to check `max_replication_slots` on all peer nodes, since there is no guarantee the configuration is the same on all of them, or that slots aren't used for other things. We'd have to do so by actually creating the slot, since otherwise someone else might create a slot and consume the last available one between when we check and when we create it.

That would force slot creation to be earlier in init, which doesn't seem desirable from the PoV of impact of join on all nodes.

I think this is more a matter of making node join progress and status more visible, so you can more easily see if it's stuck because we can't create a slot on one or more nodes.
